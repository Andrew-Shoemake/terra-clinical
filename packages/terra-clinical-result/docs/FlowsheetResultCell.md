# Terra Clinical Flowsheet Result Cell

The Clinical Result is a standardized view for single result values added to a Patient's Medical Chart. The flowsheet result cell is specifically for the grid display of a result collection, and can handle displaying an indication of multiple results.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
```
$ npm install terra-clinical-result
```

<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
<!-- AUTO-GENERATED-CONTENT:END -->

## Component Features

* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
* [LTR/RTL Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#ltr--rtl-support)

## Usage Guide

The flowsheet cell for clinical results is constructed by providing the same structured object as the clinical result in an array, allowing for multiple results to be indicated in the same time cell.

Currently the Flowsheet expects an array of one or more results - and if there are multiple, it will display the first in the array, with an indication of additional results. Updates are in progress to also look at the performedDateTime value if provided to evaluate which result is the most recent to appropriately display, regardless of which order index position in the array it is.


```jsx
const propTypes = {
  /**
   *  A set of clinical results.                                                          .
   */
  resultDataSet: PropTypes.arrayOf(PropTypes.shape({
    /**
     *  A single clinical result or blood pressure result id.
     */
    id: PropTypes.string,
    /**
     *  A single clinical result or blood pressure result.
     */
    resultData: observationPropShape,
  })),
  /**
   * Visually hides the unit of measure when presented in a series of side-by-side columns of the same unit.
   */
  hideUnit: PropTypes.bool,
  /**
   *  Display to show the full Result Name/Label Concept, e.g. `'Temperature Oral'`.
   */
  conceptDisplay: PropTypes.string,
  /**
   *  Display to show an appropriate clinically relevant documented datetime.
   */
  datetimeDisplay: PropTypes.string,
  /**
   * Whether or not the text should be truncated in display. Restricts clinical result details each to one line.
   */
  isTruncated: PropTypes.bool,
  /**
   *  If the Result value has an appended comment.
   */
  isUnverified: PropTypes.bool,
  /**
   *  If the Result value has not been authenticated and committed to patient chart.
   */
  isModified: PropTypes.bool,
  /**
   *  If the Result value has been modified from it's original value for the same clinically documented event & datetime.
   */
  hasComment: PropTypes.bool,
  /**
   * Override that shows an Error display. Used when there is a known error or problem when retrieving or assembling the clinical result data.
   */
  hasResultError: PropTypes.bool,
  /**
   * Override that shows a known "No Data" display. Used when there is known to be no value for a given clinical result concept at a specific datetime.
   */
  hasResultNoData: PropTypes.bool,
  /**
   * The padding styling to apply to the Time Column Header Cell.
   * One of `'compact'`, `'standard'`, `'none'`.
   */
  paddingStyle: PropTypes.oneOf(['compact', 'standard', 'none']),
  ```

An example of a single clinical result value:
```jsx
import React from 'react';
import { FlowsheetResultCell } from 'terra-clinical-result/lib/index';

const singleResultValue = [
  {
    eventId: '1574537491',
    result: {
      value: '101.9',
      unit: 'degC',
    },
    interpretation: 'CRITICAL',
    type: 'NUMERIC',
  },
];

export default () => (
  <FlowsheetResultCell
    resultDataSet={singleResultValue}
    hideUnit
    isModified
    hasComment
    isUnverified
  />
);
```

An example of multiple clinical blood pressure result values:
```jsx
import React from 'react';
import { FlowsheetResultCell } from 'terra-clinical-result/lib/index';
import classNames from 'classnames/bind';
import styles from '../Examples.module.scss';

const cx = classNames.bind(styles);

const multipleResultBPValues = [
  {
    id: '1577895000-1',
    systolic: {
      eventId: '1577895000-1.1',
      result: {
        value: '111',
        unit: 'mmHg',
      },
      interpretation: 'HIGH',
    },
    diastolic: {
      eventId: '1577895000-1.2',
      result: {
        value: '77',
        unit: 'mmHg',
      },
      interpretation: 'LOW',
    },
  },
  {
    id: '1577895000-2',
    systolic: {
      eventId: '1577895000-2.1',
      result: {
        value: '111',
        unit: 'mmHg',
      },
      interpretation: 'CRITICAL',
    },
    diastolic: {
      eventId: '1577895000-2.2',
      result: {
        value: '77',
        unit: 'mmHg',
      },
    },
  },
  {
    id: '1577895000-3',
    systolic: {
      eventId: '1577895000-3.1',
      result: {
        value: '111',
        unit: 'mmHg',
      },
      interpretation: 'CRITICAL',
    },
    diastolic: {
      eventId: '1577895000-3.2',
      result: {
        value: '77',
        unit: 'mmHg',
      },
    },
  },
];

export default () => (
  <FlowsheetResultCell 
    resultDataSet={multipleResultBPValues} 
    hideUnit
    isModified
    hasComment
  />
);
```
