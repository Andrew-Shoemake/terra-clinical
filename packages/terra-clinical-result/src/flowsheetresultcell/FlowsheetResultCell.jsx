import React, { useRef, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import classNames from 'classnames/bind';
import IconComment from 'terra-icon/lib/icon/IconComment';
import IconModified from 'terra-icon/lib/icon/IconModified';
import IconUnverified from 'terra-icon/lib/icon/IconDiamond';
import ClinicalResult from '../ClinicalResult';
import ClinicalResultBloodPressure from '../ClinicalResultBloodPressure';
import observationPropShape from '../proptypes/observationPropTypes';
import EnteredInError from '../common/other/_EnteredInError';
import ResultError from '../common/other/_ResultError';
import NoData from '../common/other/_KnownNoData';
import NumericOverflow from '../common/other/_NumericOverflow';
import { isEmpty, checkIsStatusInError, checkIsOfType } from '../common/utils';
import styles from './FlowsheetResultCell.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   *  A set of clinical results.                                                          .
   */
  resultDataSet: PropTypes.arrayOf(PropTypes.shape({
    /**
     *  A single clinical result or blood pressure result id.
     */
    id: PropTypes.string,
    /**
     *  A single clinical result or blood pressure result.
     */
    resultData: observationPropShape,
  })),
  /**
   * Visually hides the unit of measure when presented in a series of side-by-side columns of the same unit.
   */
  hideUnit: PropTypes.bool,
  /**
   * The padding styling to apply to the Time Column Header Cell.
   * One of `'none'`, `'standard'`, `'compact'`.
   */
  paddingStyle: PropTypes.oneOf(['none', 'standard', 'compact']),
  /**
   * Override that shows an Error display. Used when there is a known error or problem when retrieving or assembling the clinical result data.
   */
  hasResultError: PropTypes.bool,
  /**
   * Override that shows a known "No Data" display. Used when there is known to be no value for a given clinical result concept at a specific datetime.
   */
  hasResultNoData: PropTypes.bool,
  /**
   * @private
   * The intl object to be injected for translations.
   */
  intl: intlShape.isRequired,
};

const defaultProps = {
  resultDataSet: [],
  hideUnit: false,
  paddingStyle: 'compact',
  hasResultError: false,
  hasResultNoData: false,
};

const FlowsheetResultCell = (props) => {
  const {
    resultDataSet,
    hideUnit,
    paddingStyle,
    hasResultError,
    hasResultNoData,
    intl,
    ...customProps
  } = props;
  const containerDiv = useRef(null);
  const [contentWidth, setContentWidth] = useState(null);
  const [numericOverflow, setNumericOverflow] = useState(false);

  useEffect(() => {
    if (containerDiv.current && resultDataSet[0]) {
      if (checkIsOfType(resultDataSet[0], 'NUMERIC')) {
        if (!contentWidth) {
          setContentWidth(containerDiv.current.children[0].getBoundingClientRect().width);
        }
        if (containerDiv.current.getBoundingClientRect().width <= contentWidth && !numericOverflow) {
          setNumericOverflow(true);
        } else if (containerDiv.current.getBoundingClientRect().width > contentWidth) {
          setNumericOverflow(false);
        }
      }
    }
  }, [resultDataSet, contentWidth, numericOverflow]);

  let flowsheetResultCellDisplay = null;

  if (hasResultError) {
    flowsheetResultCellDisplay = <div className={cx(['primary-display', 'error'])}><ResultError /></div>;
  } else if (hasResultNoData) {
    flowsheetResultCellDisplay = <div className={cx('primary-display')}><NoData /></div>;
  } else {
    const createResultsDisplay = (resultSet) => {
      let resultKeyID;
      let resultsDisplay = [];
      let resultsInnerDisplay;
      let additionalResultCount = 0;
      let additionalResultInnerDisplay;
      const additionalResultInterpretations = [];
      let additionalResultInterpretationIndicator;
      let endAccessoryIcons;
      let primaryResultHasComment = false;
      let primaryResultIsModified = false;
      let primaryResultIsUnverified = false;
      let primaryDisplayWithInterpretation = false;

      if (!resultSet || !resultSet.length) {
        resultKeyID = 'Error';
        resultsDisplay.push(<div key={(`ClinicalResultDisplay-${resultKeyID}`)} className={cx(['primary-display', 'error'])}><ResultError /></div>);
      } else {
        resultsDisplay = [];
        additionalResultCount = resultSet.length - 1;
        for (let i = 0; i < resultSet.length; i += 1) {
          const hasStandardResult = resultSet[i].result;
          const hasSystolic = resultSet[i].systolic;
          const hasDiastolic = resultSet[i].diastolic;
          if (hasStandardResult) {
            if (i === 0) {
              const resultItem = resultSet[i];
              primaryDisplayWithInterpretation = !isEmpty(resultSet[i].interpretation);
              if (resultSet[i].hasComment) { primaryResultHasComment = true; resultItem.hasComment = false; }
              if (resultSet[i].isModified) { primaryResultIsModified = true; resultItem.isModified = false; }
              if (resultSet[i].isUnverified) { primaryResultIsUnverified = true; resultItem.isUnverified = false; }
              if (resultSet[i].eventId) resultKeyID = resultSet[i].eventId;
              else if (resultSet[i].id) resultKeyID = resultSet[i].id;
              if (checkIsStatusInError(resultSet[i])) {
                resultsInnerDisplay = <EnteredInError />;
              } else if (numericOverflow) {
                resultsInnerDisplay = <NumericOverflow />;
              } else {
                resultsInnerDisplay = (<ClinicalResult key={(`ClinicalResult-${resultKeyID}`)} resultData={resultItem} hideUnit={hideUnit} isTruncated />);
              }
            } else if (i > 0) {
              const hasInterpretation = !isEmpty(resultSet[i].interpretation) ? resultSet[i].interpretation : null;
              additionalResultInterpretations.push(hasInterpretation);
            }
          } else if (hasSystolic || hasDiastolic) {
            if (i === 0) {
              const resultItem = resultSet[i];
              const isStatusInError = (s = null, d = null) => ({
                systolic: s,
                diastolic: d,
              });
              if (hasSystolic) {
                primaryDisplayWithInterpretation = !isEmpty(resultSet[i].systolic.interpretation);
                if (resultSet[i].systolic.hasComment) { primaryResultHasComment = true; resultItem.systolic.hasComment = false; }
                if (resultSet[i].systolic.isModified) { primaryResultIsModified = true; resultItem.systolic.isModified = false; }
                if (resultSet[i].systolic.isUnverified) { primaryResultIsUnverified = true; resultItem.systolic.isUnverified = false; }
                isStatusInError.systolic = checkIsStatusInError(resultSet[i].systolic);
              }
              if (hasDiastolic) {
                if (resultSet[i].diastolic.hasComment) { primaryResultHasComment = true; resultItem.diastolic.hasComment = false; }
                if (resultSet[i].diastolic.isModified) { primaryResultIsModified = true; resultItem.diastolic.isModified = false; }
                if (resultSet[i].diastolic.isUnverified) { primaryResultIsUnverified = true; resultItem.diastolic.isUnverified = false; }
                isStatusInError.diastolic = checkIsStatusInError(resultSet[i].diastolic);
              }
              if (resultSet[i].id) resultKeyID = resultSet[i].id;
              else if (hasSystolic && resultSet[i].systolic.eventId) resultKeyID = resultSet[i].systolic.eventId;
              else if (hasDiastolic && resultSet[i].diastolic.eventId) resultKeyID = resultSet[i].diastolic.eventId;
              if (isStatusInError.systolic || isStatusInError.diastolic) {
                resultsInnerDisplay = <EnteredInError />;
              } else {
                resultsInnerDisplay = (<ClinicalResultBloodPressure key={(`ClinicalResultBloodPressure-${resultKeyID}`)} resultData={resultItem} hideUnit={hideUnit} isTruncated />);
              }
            } else if (i > 0) {
              const sysInterpretation = !isEmpty(resultSet[i].systolic.interpretation) ? resultSet[i].systolic.interpretation : null;
              const diaInterpretation = !isEmpty(resultSet[i].diastolic.interpretation) ? resultSet[i].diastolic.interpretation : null;
              additionalResultInterpretations.push(sysInterpretation);
              additionalResultInterpretations.push(diaInterpretation);
            }
          }
        }

        const primaryResultClassnames = cx([
          'primary-display',
          { unverified: primaryResultIsUnverified },
          { interpretation: primaryDisplayWithInterpretation },
        ]);

        resultsDisplay.push(<div key={(`ClinicalResultDisplay-${resultKeyID}`)} className={primaryResultClassnames} ref={containerDiv}>{resultsInnerDisplay}</div>);

        if (additionalResultInterpretations.length > 0) {
          if (additionalResultInterpretations.includes('CRITICAL')
            || additionalResultInterpretations.includes('EXTREMEHIGH')
            || additionalResultInterpretations.includes('EXTREMELOW')
            || additionalResultInterpretations.includes('PANICHIGH')
            || additionalResultInterpretations.includes('PANICLOW')
            || additionalResultInterpretations.includes('VABNORMAL')
          ) {
            additionalResultInterpretationIndicator = 'CRITICAL';
          } else if (additionalResultInterpretations.includes('POSITIVE')
            || additionalResultInterpretations.includes('ABNORMAL')
            || additionalResultInterpretations.includes('HIGH')
            || additionalResultInterpretations.includes('LOW')
          ) {
            additionalResultInterpretationIndicator = 'HIGH';
          }
        }

        const additionalResultClassNames = cx([
          'additional-end-display',
          { 'no-accessory-icons': !(primaryResultHasComment || primaryResultIsModified || primaryResultIsUnverified) },
          { 'interpretation-critical': additionalResultInterpretationIndicator === 'CRITICAL' },
          { 'interpretation-high': additionalResultInterpretationIndicator === 'HIGH' },
        ]);

        if (additionalResultCount > 0) {
          const additionalCountDisplayValue = ((additionalResultCount + 1) > 99)
            ? (<span className={cx(['additional-results-value', 'additional-results-max-value'])}>99+</span>)
            : (<span className={cx('additional-results-value')}>{additionalResultCount + 1}</span>);
          additionalResultInnerDisplay = (
            <div key={(`AdditionalResultsDisplay-${resultKeyID}`)} className={additionalResultClassNames}>
              <div
                className={cx('additional-results-stack')}
                ariaLabel={intl.formatMessage({ id: 'Terra.clinicalResult.additionalResults' }, { numberOfAdditionalResults: additionalResultCount })}
              >
                {additionalCountDisplayValue}
                {additionalCountDisplayValue}
              </div>
            </div>
          );
        }
        resultsDisplay.push(additionalResultInnerDisplay);

        const commentIconElement = primaryResultHasComment && !primaryResultIsUnverified ? (<IconComment className={cx('icon-comment')} />) : null;
        const modifiedIconElement = primaryResultIsModified && !primaryResultIsUnverified ? (<IconModified className={cx('icon-modified')} />) : null;
        const unverifiedIconElement = primaryResultIsUnverified ? (<IconUnverified className={cx('icon-unverified')} />) : null;
        if (primaryResultHasComment || primaryResultIsModified || primaryResultIsUnverified) {
          endAccessoryIcons = (
            <div key={(`EndAccessoryIcons-${resultDataSet[0].id}`)} className={cx('end-accessory-icons')}>
              <div className={cx('end-accessory-stack')}>
                {commentIconElement}
                {modifiedIconElement}
                {unverifiedIconElement}
              </div>
            </div>
          );
        } else {
          endAccessoryIcons = null;
        }
        resultsDisplay.push(endAccessoryIcons);
      }
      return resultsDisplay;
    };

    flowsheetResultCellDisplay = createResultsDisplay(resultDataSet);
  }

  const flowsheetCellClassNames = cx([
    'flowsheet-result-cell',
    { 'padding-standard': paddingStyle === 'standard' },
    { 'padding-compact': paddingStyle === 'compact' },
  ]);

  return (
    <div
      {...customProps}
      className={customProps.className ? `${flowsheetCellClassNames} ${customProps.className}` : flowsheetCellClassNames}
    >
      {flowsheetResultCellDisplay}
    </div>
  );
};

FlowsheetResultCell.propTypes = propTypes;
FlowsheetResultCell.defaultProps = defaultProps;

export default injectIntl(FlowsheetResultCell);
