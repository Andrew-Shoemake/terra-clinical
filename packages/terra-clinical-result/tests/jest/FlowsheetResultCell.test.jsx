import React from 'react';
import FlowsheetResultCell from '../../src/flowsheetresultcell/FlowsheetResultCell';
import DefaultResult, { DefaultBloodPressureResult } from '../../src/terra-dev-site/test/clinical-result/TestResults';

// Snapshot Tests
describe('FlowsheetResultCell', () => {
  it('should render when given result data', () => {
    const results = [
      DefaultResult,
      DefaultBloodPressureResult,
    ];
    const cell = shallow(<FlowsheetResultCell resultDataSet={results} />);
    expect(cell).toMatchSnapshot();
  });
  
  it('should render an error when given no result data', () => {
    const cell = shallow(<FlowsheetResultCell />);
    expect(cell).toMatchSnapshot();
  });

  it('should pass hideUnit down', () => {
    const results = [
      DefaultResult,
      DefaultBloodPressureResult,
    ];
    const cell = shallow(<FlowsheetResultCell resultDataSet={results} hideUnit />);
    expect(cell).toMatchSnapshot();
  });

  describe('paddingStyle', () => {
    it('should render when given none', () => {
      const results = [
        DefaultResult,
        DefaultBloodPressureResult,
      ];
      const cell = shallow(<FlowsheetResultCell resultDataSet={results} paddingStyle="none" />);
      expect(cell).toMatchSnapshot();
    });

    it('should render when given standard', () => {
      const results = [
        DefaultResult,
        DefaultBloodPressureResult,
      ];
      const cell = shallow(<FlowsheetResultCell resultDataSet={results} paddingStyle="standard" />);
      expect(cell).toMatchSnapshot();
    });

    it('should render when given compact', () => {
      const results = [
        DefaultResult,
        DefaultBloodPressureResult,
      ];
      const cell = shallow(<FlowsheetResultCell resultDataSet={results} paddingStyle="compact" />);
      expect(cell).toMatchSnapshot();
    });
  });
});
