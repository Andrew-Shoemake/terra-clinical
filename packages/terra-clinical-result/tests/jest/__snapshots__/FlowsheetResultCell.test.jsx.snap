// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowsheetResultCell paddingStyle should render when given compact 1`] = `
<div
  className="flowsheet-result-cell padding-compact"
>
  <div
    className="single-result-display"
    key="ClinicalResultDisplay-111"
  >
    <ClinicalResult
      hasResultError={false}
      hasResultNoData={false}
      hideUnit={false}
      isTruncated={true}
      key="ClinicalResult-111"
      resultData={
        Object {
          "eventId": "111",
          "result": Object {
            "unit": "mL",
            "value": "12345.678",
          },
        }
      }
    />
  </div>
  <div
    className="additional-results-display add-end-accessory-space"
    key="AdditionalResultsDisplay-111"
  >
    <div
      className="additional-results-stack"
    >
      <span
        className="additional-results-value"
      >
        2
      </span>
      <span
        className="additional-results-value"
      >
        2
      </span>
    </div>
  </div>
</div>
`;

exports[`FlowsheetResultCell paddingStyle should render when given none 1`] = `
<div
  className="flowsheet-result-cell"
>
  <div
    className="single-result-display"
    key="ClinicalResultDisplay-111"
  >
    <ClinicalResult
      hasResultError={false}
      hasResultNoData={false}
      hideUnit={false}
      isTruncated={true}
      key="ClinicalResult-111"
      resultData={
        Object {
          "eventId": "111",
          "result": Object {
            "unit": "mL",
            "value": "12345.678",
          },
        }
      }
    />
  </div>
  <div
    className="additional-results-display add-end-accessory-space"
    key="AdditionalResultsDisplay-111"
  >
    <div
      className="additional-results-stack"
    >
      <span
        className="additional-results-value"
      >
        2
      </span>
      <span
        className="additional-results-value"
      >
        2
      </span>
    </div>
  </div>
</div>
`;

exports[`FlowsheetResultCell paddingStyle should render when given standard 1`] = `
<div
  className="flowsheet-result-cell padding-standard"
>
  <div
    className="single-result-display"
    key="ClinicalResultDisplay-111"
  >
    <ClinicalResult
      hasResultError={false}
      hasResultNoData={false}
      hideUnit={false}
      isTruncated={true}
      key="ClinicalResult-111"
      resultData={
        Object {
          "eventId": "111",
          "result": Object {
            "unit": "mL",
            "value": "12345.678",
          },
        }
      }
    />
  </div>
  <div
    className="additional-results-display add-end-accessory-space"
    key="AdditionalResultsDisplay-111"
  >
    <div
      className="additional-results-stack"
    >
      <span
        className="additional-results-value"
      >
        2
      </span>
      <span
        className="additional-results-value"
      >
        2
      </span>
    </div>
  </div>
</div>
`;

exports[`FlowsheetResultCell should pass hideUnit down 1`] = `
<div
  className="flowsheet-result-cell padding-compact"
>
  <div
    className="single-result-display"
    key="ClinicalResultDisplay-111"
  >
    <ClinicalResult
      hasResultError={false}
      hasResultNoData={false}
      hideUnit={true}
      isTruncated={true}
      key="ClinicalResult-111"
      resultData={
        Object {
          "eventId": "111",
          "result": Object {
            "unit": "mL",
            "value": "12345.678",
          },
        }
      }
    />
  </div>
  <div
    className="additional-results-display add-end-accessory-space"
    key="AdditionalResultsDisplay-111"
  >
    <div
      className="additional-results-stack"
    >
      <span
        className="additional-results-value"
      >
        2
      </span>
      <span
        className="additional-results-value"
      >
        2
      </span>
    </div>
  </div>
</div>
`;

exports[`FlowsheetResultCell should render an error when given no result data 1`] = `
<div
  className="flowsheet-result-cell padding-compact"
>
  <div
    className="single-result-display "
    key="ClinicalResultDisplay-Error"
  >
    <ResultError />
  </div>
</div>
`;

exports[`FlowsheetResultCell should render when given result data 1`] = `
<div
  className="flowsheet-result-cell padding-compact"
>
  <div
    className="single-result-display"
    key="ClinicalResultDisplay-111"
  >
    <ClinicalResult
      hasResultError={false}
      hasResultNoData={false}
      hideUnit={false}
      isTruncated={true}
      key="ClinicalResult-111"
      resultData={
        Object {
          "eventId": "111",
          "result": Object {
            "unit": "mL",
            "value": "12345.678",
          },
        }
      }
    />
  </div>
  <div
    className="additional-results-display add-end-accessory-space"
    key="AdditionalResultsDisplay-111"
  >
    <div
      className="additional-results-stack"
    >
      <span
        className="additional-results-value"
      >
        2
      </span>
      <span
        className="additional-results-value"
      >
        2
      </span>
    </div>
  </div>
</div>
`;
